src/delay.rs

```diff
@@ -54,10 +61,7 @@ impl From<TickType> for TickType_t {
 
 impl From<Duration> for TickType {
     fn from(duration: Duration) -> Self {
-        TickType(
-            ((duration.as_millis() + TICK_PERIOD_MS as u128 - 1) / TICK_PERIOD_MS as u128)
-                as TickType_t,
-        )
+        TickType::new_millis(min(duration.as_millis(), u64::MAX as _) as _)
```

Comment:
Turns out `Duration::as_millis` is rounding down. Perhaps we want to address this?

Reply:
Yes, I agree. We should address this.